Integrate KafkaCleaner

KillianG opened the pull request:
Issue: ZENKO-4617

<!--
Thank you for contributing to Zenko!

Please enter applicable information below.
-->

**What does this PR do, and why do we need it?**

**Which issue does this PR fix?**

fixes #<ISSUE>

**Special notes for your reviewers**:

Commit:
Integrate KafkaCleaner

Issue: ZENKO-4617

bert-e commented:
# Hello killiang,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4617 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.44`

* `2.7.40`

* `2.8.20`

* `2.9.0`


Please check the `Fix Version/s` of ZENKO-4617, or the target
branch of this pull request.

Commit:
fixup! nothing

Commit:
Issue: {Jira_ticket}

Commit:
Issue: {Jira_ticket}

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/ZENKO-4617-integrate-kafka-cleaner` has diverged from
`development/2.7` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.7` into `improvement/ZENKO-4617-integrate-kafka-cleaner`
* Rebase `improvement/ZENKO-4617-integrate-kafka-cleaner` onto `origin/development/2.7`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

KillianG commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/ZENKO-4617-integrate-kafka-cleaner` has diverged from
`development/2.7` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.7` into `improvement/ZENKO-4617-integrate-kafka-cleaner`
* Rebase `improvement/ZENKO-4617-integrate-kafka-cleaner` onto `origin/development/2.7`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

KillianG commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/ZENKO-4617-integrate-kafka-cleaner` has diverged from
`development/2.7` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.7` into `improvement/ZENKO-4617-integrate-kafka-cleaner`
* Rebase `improvement/ZENKO-4617-integrate-kafka-cleaner` onto `origin/development/2.7`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

KillianG commented:
/reset

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4617 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.7.41`

* `2.8.21`

* `2.9.0`


Please check the `Fix Version/s` of ZENKO-4617, or the target
branch of this pull request.

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

KillianG commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-4617-integrate-kafka-cleaner` into
`development/2.7`
* [w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.8...w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.8` 
* [w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.9...w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

Commit:
Configure kafkacleaner to run more often

Change the interval of kafkacleaner runs to 30s to make it testable

Issue: ZENKO-4617

Commit:
Add tests for kafka-cleaner

After adding kafka cleaner we can now test that it's really working well
After existing restore and archive tests we now check for the remaining messages in the topic to be sure kafka cleaner deleted them correctly

Issue: ZENKO-4617

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

Please resolve the conflict on **the integration branch** (`w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ git pull  # or "git reset --hard origin/w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner"
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

KillianG commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #8c8d79a6ba353bfd70ae5924f5437a6d55e91dd9 on the integration branch
`w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner` is merging a branch which is neither the current
branch `improvement/ZENKO-4617-integrate-kafka-cleaner` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `improvement/ZENKO-4617-integrate-kafka-cleaner` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

Commit:
Add kafkajs to ctst

Issue: ZENKO-4617

Commit:
Test kafkacleaner

Test that kafka cleaner correctly remove consumed messages from all topics apart from oplog and dead letter queue ones

Issue: ZENKO-4617

KillianG commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **create_integration_branches**

KillianG commented:
/force_reset

bert-e commented:
# History mismatch



Merge commit #8c8d79a6ba353bfd70ae5924f5437a6d55e91dd9 on the integration branch
`w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner` is merging a branch which is neither the current
branch `improvement/ZENKO-4617-integrate-kafka-cleaner` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `improvement/ZENKO-4617-integrate-kafka-cleaner` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

KillianG commented:
/create_integration_branches

Commit:
Remove unused world params

Issue: ZENKO-4617

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-4617-integrate-kafka-cleaner` into
`development/2.7`
* [w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.8...w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.8` 
* [w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.9...w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

Commit:
add AfterAll tag

Issue: ZENKO-4617

bert-e commented:
# History mismatch



Merge commit #12263ecd55e8fdad4cae03049cbf79f5c0c1a3cc on the integration branch
`w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` is merging a branch which is neither the current
branch `improvement/ZENKO-4617-integrate-kafka-cleaner` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `improvement/ZENKO-4617-integrate-kafka-cleaner` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

KillianG commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #12263ecd55e8fdad4cae03049cbf79f5c0c1a3cc on the integration branch
`w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` is merging a branch which is neither the current
branch `improvement/ZENKO-4617-integrate-kafka-cleaner` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `improvement/ZENKO-4617-integrate-kafka-cleaner` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

KillianG commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-4617-integrate-kafka-cleaner` into
`development/2.7`
* [w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.8...w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.8` 
* [w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.9...w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #d9b922cc9faaacbe0d3f815e5cec12c153646a10 on the integration branch
`w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` is merging a branch which is neither the current
branch `improvement/ZENKO-4617-integrate-kafka-cleaner` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `improvement/ZENKO-4617-integrate-kafka-cleaner` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

KillianG commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner origin/development/2.8
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-4617-integrate-kafka-cleaner` into
`development/2.7`
* [w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.8...w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.8` 
* [w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner](https://github.com/scality/zenko/compare/development/2.9...w/2.9/improvement/ZENKO-4617-integrate-kafka-cleaner) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

Commit:
Remove unbound variable

Issue: ZENKO-4617

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner` with contents from `improvement/ZENKO-4617-integrate-kafka-cleaner`
and `development/2.8`.

Please resolve the conflict on **the integration branch** (`w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ git pull  # or "git reset --hard origin/w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner"
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/ZENKO-4617-integrate-kafka-cleaner
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/ZENKO-4617-integrate-kafka-cleaner
```





*The following options are set:* **create_integration_branches**

Commit:
Adding log to ease debugging

Issue: ZENKO-4617

Commit:
Missing semicolon

Issue: ZENKO-4617

Commit:
KafkaCleaner check wait more

Wait for the cron rule to apply also when waiting for kfkc to pass to avoid flakyness

Issue: ZENKO-4617

Commit:
Checking cleanre works for bucket tasks

Issue: ZENKO-4617

Commit:
Add comment

Add comment to explain why we dont check strictly for the equality for bucket-tasks

issue: ZENKO-4617

Commit:
Refactor function

Issue: ZENKO-4617

Commit:
Parseint

issue: ZENKO-4617

Commit:
Better
